drgn_tools/crash_net.py

```diff
@@ -0,0 +1,372 @@
+        + "    CPU: "
+        + str(int(task.cpu))
+        + "    COMMAND: "
+        + task.comm.string_().decode("utf-8")
+    )
+
+    if print_full_data:
+        print(
+            "%-5s %-20s %-20s %-20s %-20s %-20s"
+            % (
+                "FD",
+                "SOCKET",
+                "SOCK",
+                "FAMILY:TYPE",
+                "SOURCE-PORT",
+                "DESTINATION-PORT",
+            )
+        )
+
+    rows = [
+        [
+            "FD",
+            "SOCKET",
+            "SOCK",
+            "FAMILY:TYPE",
+            "SOURCE-PORT",
+            "DESTINATION-PORT",
+        ]
+    ]
+
+    for i in range(int(task.files.fdt.max_fds)):
+        try:
+            socket = SOCKET_I(task.files.fdt.fd[i].f_path.dentry.d_inode)
+            sock_family = str(
+                sockfamily_dict.get(int(socket.sk.__sk_common.skc_family))
+            )
+            if (
+                int(socket.sk.__sk_common.skc_family)
+            ) not in sockfamily_dict.keys():
+                sock_family = "AF_"
+            sock_family = sock_family[3:]
+            sock_type = str(socktype_dict.get(int(socket.type)))
+            if int(socket.type) not in socktype_dict.keys():
+                sock_type = "SOCK_"
+            sock_type = sock_type[5:]
+
+            if sock_family == "INET":
+                saddr = int(socket.sk.__sk_common.skc_rcv_saddr)
+                daddr = int(socket.sk.__sk_common.skc_daddr)
+                dport = socketlib.ntohs(int(socket.sk.__sk_common.skc_dport))
+                sport = int(socket.sk.__sk_common.skc_num)
+                source_port = (
+                    str(ipaddress.ip_address(saddr).reverse_pointer[:-13])
+                    + "-"
+                    + str(sport)
+                )
+                destination_port = (
+                    ipaddress.ip_address(daddr).reverse_pointer[:-13]
+                    + "-"
+                    + str(dport)
+                )
+            else:
+                source_port = ""
+                destination_port = ""
+            rows.append(
+                [
+                    str(i),
+                    hex(socket.value_())[2:],
+                    hex(socket.sk.value_())[2:],
+                    sock_family + ":" + sock_type,
+                    source_port,
+                    destination_port,
+                ]
+            )
+            if print_full_data:
+                print(
+                    "%-5s %-20s %-20s %-20s %-20s %-20s"
+                    % (
+                        str(i),
+                        hex(socket.value_())[2:],
+                        hex(socket.sk.value_())[2:],
+                        sock_family + ":" + sock_type,
+                        source_port,
+                        destination_port,
+                    )
+                )
+                print(socket)
+                print(socket.sk)
+
+        except Exception:
+            continue
```

Comment:
Copying over an unresolved comment from v1:

This `try/except` block spans 40 or 50 lines of code. I can't tell what error it's supposed to catch. And given that the `except` block catches `Exception`... well, anything could happen here and this function would just swallow errors and continue on.

In general, `try/except` blocks should either:
1. Be small, so that the reader can understand exactly where the expected error could come from, and why. Clearly, this try/except block is too big for that.
2. Serve some important error resilience purpose (for example: `corelens` catches all exceptions raised by a `CorelensModule` and logs them, but then continues to run the remaining modules -- that way corelens is resilient to logical errors in modules).

In this case, resilience isn't as important as correctness. If we can't properly format information about a socket, this function should crash, so that we can learn why and fix the function.

---

It seems to me that the main error you may care about is that the file is not corresponding to a socket. According to the [docs](https://drgn.readthedocs.io/en/latest/helpers.html#drgn.helpers.linux.net.SOCKET_I) for `SOCKET_I`, this raises a `ValueError` in that case. So if that's all you care about, you could do:

```python
try:
    socket = SOCKET_I(file)
except ValueError:
    # file is not a socket
    continue
```